shoop/admin/forms/widgets.py

```diff
@@ -75,14 +75,17 @@ def render(self, name, value, attrs=None):
         if self.clearable:
             bits.insert(1, self.get_clear_markup())
 
+        browser_attrs = {
+            "class": "browse-widget %s-browse-widget" % self.browse_kind,
+            "data-browse-kind": self.browse_kind,
+            "data-clearable": self.clearable,
+            "data-empty-text": self.empty_text
+        }
+        if self.filter:
+            browser_attrs["data-filter"] = self.filter
+
         return mark_safe("<div %(attrs)s>%(content)s</div>" % {
-            "attrs": flatatt({
-                "class": "browse-widget %s-browse-widget" % self.browse_kind,
-                "data-browse-kind": self.browse_kind,
-                "data-clearable": self.clearable,
-                "data-empty-text": self.empty_text,
-                "data-filter": self.filter
-            }),
+            "attrs": flatatt(browser_attrs),
```

Comment:
It'd be better to just use `flatatt_filter` (in `shoop.admin.toolbar`), which automatically elides falsy values.